pyanaconda/modules/timezone/geoloc.py

```diff
@@ -0,0 +1,171 @@
+#
+# Copyright (C) 2022  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+from abc import abstractmethod
+import time
+import requests
+from pyanaconda.core.util import requests_session
+from pyanaconda.core import constants
+from pyanaconda.timezone import get_preferred_timezone, is_valid_timezone
+from pyanaconda.modules.common.structures.timezone import GeolocationResult
+
+from pyanaconda.anaconda_loggers import get_module_logger, get_sensitive_info_logger
+log = get_module_logger(__name__)
+sensitive_info_log = get_sensitive_info_logger()
+
+
+class GeolocationBackend:
+    """Base class for GeoIP backends."""
+    def __init__(self):
+        self._result = GeolocationResult()
+        self._session = requests_session()
```

Comment:
This should be eventually used as `with requests_session() as session:`.

Reply:
Even better, there is just a single request to a http target, so no need for a session...